### DISCUSSION TITLE
Row selection, can only delete one row.

### DISCUSSION THREAD

**Discussion by jacobmaussocc** [Budibase/budibase#11559] [Help]
It seems like the row selection deletion only works if it has one row. It seems like it stores all of the different rows in a string separated by commas, I think I may try to write a javascript automation that can do this functionality, I just wanted to know if there is an easier solution.

**Comment by ConorWebb96** [comment]
Hey @jacobmaussocc,

I may not fully understand what you are asking, hopefully the below answers your question.

You should be able to just supply the `Delete row` action with the `Selected rows` binding. This automatically creates the ids separated by commas for you.

Below ill show you a working example of this in action. Ignore that I'm using the row on click to use this action. Normally this would be used on a button's click.

The whole setup for the table + row selection. (Make sure the row selection checkbox is ticket.)
<img width="1362" alt="Screenshot 2023-08-21 at 07 42 46" src="https://github.com/Budibase/budibase/assets/126772285/3f859d70-3de2-4522-a1cc-e5cf9e403f26">

**Output**
<img width="1354" alt="Screenshot 2023-08-21 at 08 23 10" src="https://github.com/Budibase/budibase/assets/126772285/76a7d7be-05da-4be1-a5e7-22047da20248">
<img width="1316" alt="Screenshot 2023-08-21 at 08 23 17" src="https://github.com/Budibase/budibase/assets/126772285/747150db-bb01-4c2a-b3f9-e08a46d86f80">

I hope this helps, with this you won't need to write Js for this.

**Reply by jacobmaussocc** [reply]
I am not sure if it is because I ejected the table block, but I tried that exact functionality and it did not work, it seemed like it couldn't read the comma separated string.

**Reply by ConorWebb96** [reply]
Hey @jacobmaussocc,

Could you provide some screenshots of your component tree? As well as your table component and the setup were you are calling the `Delete row` action?

**Comment by jacobmaussocc** [comment]
![image](https://github.com/Budibase/budibase/assets/136478885/5ed2ac71-9b13-43ee-8e7c-8d1aaa4cf373)
This is the error I receive when deleting multiple rows. Here I actually was able to test it with a regular block and it still won't work.
![image](https://github.com/Budibase/budibase/assets/136478885/f3d4d4d4-f74f-4265-8a7e-33ae4b1f1d70)
![image](https://github.com/Budibase/budibase/assets/136478885/0c16e5ed-e4a8-4337-8053-7328c4e5f1c3)
![image](https://github.com/Budibase/budibase/assets/136478885/7f7069fe-b1a9-456c-a78d-ab7bd3629244)
![image](https://github.com/Budibase/budibase/assets/136478885/41704708-d32c-4f9c-839e-fa4cb3f40b72)

**Reply by jacobmaussocc** [reply]
If it helps I also have a screenshot of what the actual info in the binding reads.
![image](https://github.com/Budibase/budibase/assets/136478885/aacf5901-ca9f-4b9a-a0b4-2c943c8d77e1)

**Reply by ConorWebb96** [reply]
Ahh! You are using an external data source. The examples I showed were using the internal database. 

I initially thought it was maybe related to the encoded _ids. But I have tested and it seems to work fine for me with Postgres and MySQL. 

Is there any chance you could create a paragraph component and output all the selected row _ids? I saw that you showed 2 but was this all of them? The undefined error you showed, tells me that one may not be set for one of your rows, which might be the cause of the issue.

Thanks in advance for the information!

**Comment by jacobmaussocc** [comment]
It was all of them, I did it again for you here. Thanks for all of your help!
![image](https://github.com/Budibase/budibase/assets/136478885/a52de1b9-8853-4e73-8f8e-4efd6103d37b)
We are using MySQL

**Reply by ConorWebb96** [reply]
Hey @jacobmaussocc,

As I can see you are blurring out information I'm going to assume its sensitive. Could you possibly recreate with a dummy db, then send over the exported app as well as the database's schema to **support@budibase.com** so I could have a more in-depth look.

You should be able to get the schema within Budibase, below are instructions of were to go to get it
1. Click the data source you wish to download the schema of
<img width="265" alt="Screenshot 2023-08-22 at 14 17 13" src="https://github.com/Budibase/budibase/assets/126772285/3345bba5-7a68-40d8-95cb-b2af0f0a4c80">

2. Click the settings tab and then click the download button.
<img width="837" alt="Screenshot 2023-08-22 at 14 14 50" src="https://github.com/Budibase/budibase/assets/126772285/fdbfb159-ade0-488c-baef-2889e844a64a">

Thanks ahead of time!

**Reply by jacobmaussocc** [reply]
I cannot figure out how to save my database schema, I just have a configuration page when I click on my new test dummy database. I have a free version of budibase if that changes anything. Thanks.

**Reply by ConorWebb96** [reply]
Hey @jacobmaussocc,

All you need to do is click on your data source like shown in the first bullet point, then click the **settings** tab and then click the download button. This will automatically download your schema for your database. Hopefully the gif below helps show you were to go!

![Screen Recording 2023-08-23 at 07 52 43](https://github.com/Budibase/budibase/assets/126772285/d310e283-bfc9-471c-ae8a-a71b37d669f0)

**Reply by jacobmaussocc** [reply]
I emailed budibase support with a gif on how I am not seeing the same stuff you are.

**Reply by ConorWebb96** [reply]
I've responded to your email. 

Issue logged here; https://github.com/Budibase/budibase/issues/11588

For anyone who stumbles on this thread. It is recommend that you don't use composite keys for primary keys. Instead either use auto incremented primary keys or completely uniquely generated primary keys.

**Reply by jacobmaussocc** [reply]
Connor, I apologize for not closely looking at this faster, I had accidentally configured the database I had sent to you with composite keys. This functionality still does not work for me when using primary keys. I had an example earlier... 
![image](https://github.com/Budibase/budibase/assets/136478885/5534a50a-d827-40e1-ad35-98cbfecfaf92)
this was not able to work and is using the primary key "row ID" for this table. I will be sending you a more accurate schema. Sorry I sent you something inaccurate.
